core/templates/components/button-directives/create-activity-button.component.spec.ts

```diff
@@ -0,0 +1,285 @@
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+
+import { ExplorationCreationService } from 'components/entity-creation-services/exploration-creation.service';
+import { TranslatePipe } from 'filters/translate.pipe';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { UserService } from 'services/user.service';
+import { CreateActivityButtonComponent } from './create-activity-button.component';
+
+/**
+ * @fileoverview Unit tests for CreateActivityButtonComponent.
+ */
+
+class MockWindowRef {
+  _window = {
+    location: {
+      _hash: '',
+      _hashChange: null,
+      _href: '',
+      get hash() {
+        return this._hash;
+      },
+      set hash(val) {
+        this._hash = val;
+        if (this._hashChange === null) {
+          return;
+        }
+        this._hashChange();
+      },
+      get href() {
+        return this._href;
+      },
+      set href(val) {
+        this._href = val;
+      },
+      reload: (val) => val,
+      replace: (val) => {}
+    },
+    get onhashchange() {
+      return this.location._hashChange;
+    },
+
+    set onhashchange(val) {
+      this.location._hashChange = val;
+    }
+  };
+  get nativeWindow() {
+    return this._window;
+  }
+}
+
+interface UrlParamsType {
+  [param: string]: string
+}
+
+class MockUrlService {
+  getPathname(): string {
+    return '/creator-dashboard';
+  }
+
+  getUrlParams(): UrlParamsType {
+    return {
+      mode: 'create'
+    };
+  }
+}
+
+describe('CreateActivityButtonComponent', () => {
+  let component: CreateActivityButtonComponent;
+  let fixture: ComponentFixture<CreateActivityButtonComponent>;
+  let userService: UserService;
+  let urlService: MockUrlService;
+  let explorationCreationService: ExplorationCreationService;
+  let windowRef: MockWindowRef;
+  let ngbModal: NgbModal;
+
+  beforeEach(waitForAsync(() => {
+    windowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      declarations: [
+        CreateActivityButtonComponent,
+        TranslatePipe
+      ],
+      providers: [
+        UserService,
+        {
+          provide: UrlService,
+          useClass: MockUrlService
+        },
+        {
+          provide: WindowRef,
+          useValue: windowRef
+        },
+        ExplorationCreationService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateActivityButtonComponent);
+    component = fixture.componentInstance;
+    userService = TestBed.inject(UserService);
+    urlService = TestBed.inject(UrlService);
+    explorationCreationService = TestBed.inject(ExplorationCreationService);
+    ngbModal = TestBed.inject(NgbModal);
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeDefined();
+  });
+
+  it('should initialize and call initCreationProcess()', fakeAsync(() => {
```

Comment:
Need to work on the names... Don't try to name it based on the code... Try to think about it more like a user. What this checks is basically that all the initializations happen.

Maybe rename this to:
should initialize component properties and then begin creation process

(Try to work on the names for the other suites, and I'll take a look again)

Reply:
Renamed the tests, PTAL!